analytics.go

```diff
@@ -57,7 +57,7 @@ type client struct {
 
 	// This HTTP client is used to send requests to the backend, it uses the
 	// HTTP transport provided in the configuration.
-	http http.Client
+	http *http.Client
 }
```

Comment:
There's one memory allocation done for the `analytics.client` object, if the `http.Client` is a value, the memory for this object will be part of the `analytics.client`. It it's a pointer Go needs to make another memory allocation for the `http.Client` and the `analytics.client` then references this other memory area, the GC then now has to manage 2 pointers intstead of one. I agree it's not a big deal in this case, it's just a good practice.

I feel like using pointers vs values allows to express ownership. When an object is gonna be shared, using a pointer makes sense because different part of the application a referencing the same object. When there's only one owner, you can get away with just a value and get better memory management.
Obviously in Go you can choose to get the address of pretty much anything and start creating shared ownership of a value, even if it wasn't declared as a pointer initially, the only downside is it could result in preventing a larger block of memory from being released (think if the object is part of a slice or struct), but usually you don't need to care about this and can just fallback to using a value.

Using pointers also have runtime small runtime costs besides putting more work on the GC because the language does nil-checking every time a pointe is dereferenced, with a value the compiler knows that the value can't ever be nil so it can optimize this away.

Other use-cases for pointers are to describe that the value is allowed to be missing (like with the `omitempty` flag in the json package), but in our case we don't ever want the http client to not exist either.